downloader/src/main/kotlin/vcs/Git.kt

```diff
@@ -183,17 +182,13 @@ class Git : VersionControlSystem(), CommandLineTool {
     private fun updateWorkingTreeWithoutSubmodules(workingTree: WorkingTree, revision: String): Result<String> =
         runCatching {
             log.info { "Trying to fetch only revision '$revision' with depth limited to $GIT_HISTORY_DEPTH." }
-            val fetch = workingTree.runGit("fetch", "--depth", "$GIT_HISTORY_DEPTH", "origin", revision)
+            workingTree.runGit("fetch", "--depth", "$GIT_HISTORY_DEPTH", "origin", revision)
 
             // The documentation for git-fetch states that "By default, any tag that points into the histories being
             // fetched is also fetched", but that is not true for shallow fetches of a tag; then the tag itself is
-            // not fetched. So create it manually afterwards.
-            val tag = fetch.stderr.lineSequence().mapNotNull {
-                it.withoutPrefix(" * tag ")?.substringBefore("->")?.trim()
-            }.firstOrNull()
-
-            if (revision == tag) {
-                workingTree.runGit("tag", revision, "FETCH_HEAD")
+            // not fetched. So create the local tag manually afterwards.
+            if (revision in workingTree.listRemoteTags()) {
+                workingTree.runGit("tag", "-f", revision, "FETCH_HEAD")
```

Comment:
Do you want to mention your rational why the `-f` flat is required in the comment above?
IMO it's important enough to not _only_ be in the commit message.

Reply:
> IMO it's important enough to not _only_ be in the commit message.

I think it's fine 😛 More seriously, it's temporary eventually anyway, as the `-f` could be removed once scan-by-repo is implemented.